In [1]:
import pandas as pd
import numpy as np

import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

pd.set_option('display.max_columns', None)

mpl.rcParams['axes.unicode_minus'] = False
plt.rcParams['font.family'] = 'Malgun Gothic'


pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)

In [2]:
df = pd.read_csv('./1년 1시간별 Ls, Ws.csv', encoding = 'cp949')
df

자치구           대여소명  대여소번호     년  월  시간    lambda        mu    c  \
0       중랑구  (구)신한은행 중랑교지점   1442  2020  6   0  3.800000  2.700000  6.0   
1       중랑구  (구)신한은행 중랑교지점   1442  2020  6   1  3.700000  2.133333  6.0   
2       중랑구  (구)신한은행 중랑교지점   1442  2020  6   2  2.133333  1.266667  6.0   
3       중랑구  (구)신한은행 중랑교지점   1442  2020  6   3  1.566667  0.833333  6.0   
4       중랑구  (구)신한은행 중랑교지점   1442  2020  6   4  1.333333  0.533333  6.0   
...     ...            ...    ...   ... ..  ..       ...       ...  ...   
694939  강서구        힐스테이트에코   2720  2021  5  19  7.933333  8.066667  9.0   
694940  강서구        힐스테이트에코   2720  2021  5  20  5.166667  7.800000  9.0   
694941  강서구        힐스테이트에코   2720  2021  5  21  3.400000  6.933333  9.0   
694942  강서구        힐스테이트에코   2720  2021  5  22  1.800000  3.000000  9.0   
694943  강서구        힐스테이트에코   2720  2021  5  23  1.233333  1.166667  9.0   

             rho       p_0       p_1      p_1+       p_c       L_s        W_s  
0       1.407407  0.040995  0.057696  0.959005  0.318602  4.249820   1.641292  
1       1.734375  0.015893  0.027565  0.984107  0.432587  4.789792   2.281475  
2       1.684211  0.018275  0.030779  0.981725  0.417101  4.725431   3.800050  
3       1.880000  0.010731  0.020174  0.989269  0.473793  4.948997   6.003217  
4       2.500000  0.002462  0.006154  0.997538  0.600985  5.344821  10.046270  
...          ...       ...       ...       ...       ...       ...        ...  
694939  0.983471  0.107665  0.105886  0.892335  0.092668  4.362561   0.606066  
694940  0.662393  0.343188  0.227325  0.656812  0.008425  1.796723   0.350708  
694941  0.490385  0.510026  0.250109  0.489974  0.000836  0.954216   0.280887  
694942  0.600000  0.402433  0.241460  0.597567  0.004056  1.439166   0.802792  
694943  1.057143  0.076892  0.081286  0.923108  0.126790  4.956086   4.601924  

[694944 rows x 16 columns]

In [3]:
df = df[df['대여소명'] == '천호역4번출구(현대백화점)']
df

자치구            대여소명  대여소번호     년   월  시간     lambda         mu     c  \
613152  강동구  천호역4번출구(현대백화점)   1009  2020   6   0   4.333333   4.700000  14.0   
613153  강동구  천호역4번출구(현대백화점)   1009  2020   6   1   3.000000   3.033333  14.0   
613154  강동구  천호역4번출구(현대백화점)   1009  2020   6   2   3.366667   3.166667  14.0   
613155  강동구  천호역4번출구(현대백화점)   1009  2020   6   3   1.866667   1.966667  14.0   
613156  강동구  천호역4번출구(현대백화점)   1009  2020   6   4   1.133333   1.233333  14.0   
613157  강동구  천호역4번출구(현대백화점)   1009  2020   6   5   1.400000   0.900000  14.0   
613158  강동구  천호역4번출구(현대백화점)   1009  2020   6   6   3.633333   1.500000  14.0   
613159  강동구  천호역4번출구(현대백화점)   1009  2020   6   7   6.166667   2.233333  14.0   
613160  강동구  천호역4번출구(현대백화점)   1009  2020   6   8   9.333333   5.366667  14.0   
613161  강동구  천호역4번출구(현대백화점)   1009  2020   6   9   7.466667   5.133333  14.0   
613162  강동구  천호역4번출구(현대백화점)   1009  2020   6  10   6.533333   3.533333  14.0   
613163  강동구  천호역4번출구(현대백화점)   1009  2020   6  11   5.300000   5.266667  14.0   
613164  강동구  천호역4번출구(현대백화점)   1009  2020   6  12   6.366667   7.633333  14.0   
613165  강동구  천호역4번출구(현대백화점)   1009  2020   6  13   7.800000   8.533333  14.0   
613166  강동구  천호역4번출구(현대백화점)   1009  2020   6  14   8.866667   8.666667  14.0   
613167  강동구  천호역4번출구(현대백화점)   1009  2020   6  15   8.833333  10.766667  14.0   
613168  강동구  천호역4번출구(현대백화점)   1009  2020   6  16  10.300000  11.533333  14.0   
613169  강동구  천호역4번출구(현대백화점)   1009  2020   6  17  12.500000  13.666667  14.0   
613170  강동구  천호역4번출구(현대백화점)   1009  2020   6  18  15.666667  18.633333  14.0   
613171  강동구  천호역4번출구(현대백화점)   1009  2020   6  19  14.466667  15.166667  14.0   
613172  강동구  천호역4번출구(현대백화점)   1009  2020   6  20  12.266667  12.366667  14.0   
613173  강동구  천호역4번출구(현대백화점)   1009  2020   6  21   9.933333   9.833333  14.0   
613174  강동구  천호역4번출구(현대백화점)   1009  2020   6  22   6.866667   7.100000  14.0   
613175  강동구  천호역4번출구(현대백화점)   1009  2020   6  23   4.700000   4.233333  14.0   
613176  강동구  천호역4번출구(현대백화점)   1009  2020   7   0   4.000000   4.000000  14.0   
613177  강동구  천호역4번출구(현대백화점)   1009  2020   7   1   3.733333   3.833333  14.0   
613178  강동구  천호역4번출구(현대백화점)   1009  2020   7   2   3.400000   3.333333  14.0   
613179  강동구  천호역4번출구(현대백화점)   1009  2020   7   3   1.566667   1.600000  14.0   
613180  강동구  천호역4번출구(현대백화점)   1009  2020   7   4   1.100000   1.366667  14.0   
613181  강동구  천호역4번출구(현대백화점)   1009  2020   7   5   1.433333   1.033333  14.0   
613182  강동구  천호역4번출구(현대백화점)   1009  2020   7   6   4.033333   1.633333  14.0   
613183  강동구  천호역4번출구(현대백화점)   1009  2020   7   7   4.500000   3.233333  14.0   
613184  강동구  천호역4번출구(현대백화점)   1009  2020   7   8   6.233333   4.866667  14.0   
613185  강동구  천호역4번출구(현대백화점)   1009  2020   7   9   4.933333   4.733333  14.0   
613186  강동구  천호역4번출구(현대백화점)   1009  2020   7  10   4.066667   3.100000  14.0   
613187  강동구  천호역4번출구(현대백화점)   1009  2020   7  11   5.066667   4.600000  14.0   
613188  강동구  천호역4번출구(현대백화점)   1009  2020   7  12   5.400000   6.000000  14.0   
613189  강동구  천호역4번출구(현대백화점)   1009  2020   7  13   4.866667   6.866667  14.0   
613190  강동구  천호역4번출구(현대백화점)   1009  2020   7  14   5.633333   5.600000  14.0   
613191  강동구  천호역4번출구(현대백화점)   1009  2020   7  15   8.333333   8.266667  14.0   
613192  강동구  천호역4번출구(현대백화점)   1009  2020   7  16   8.433333   9.033333  14.0   
613193  강동구  천호역4번출구(현대백화점)   1009  2020   7  17   9.866667  11.566667  14.0   
613194  강동구  천호역4번출구(현대백화점)   1009  2020   7  18  12.700000  13.133333  14.0   
613195  강동구  천호역4번출구(현대백화점)   1009  2020   7  19  12.433333  12.166667  14.0   
613196  강동구  천호역4번출구(현대백화점)   1009  2020   7  20   9.100000   9.933333  14.0   
613197  강동구  천호역4번출구(현대백화점)   1009  2020   7  21   8.400000   8.666667  14.0   
613198  강동구  천호역4번출구(현대백화점)   1009  2020   7  22   6.800000   6.300000  14.0   
613199  강동구  천호역4번출구(현대백화점)   1009  2020   7  23   5.333333   5.466667  14.0   
613200  강동구  천호역4번출구(현대백화점)   1009  2020   8   0   1.566667   1.200000  14.0   
613201  

In [4]:
df = df.drop(['대여소번호', 'lambda', 'mu', 'c', 'rho', 'p_c', 'p_1+'], axis = 1)
df

자치구            대여소명     년   월  시간           p_0           p_1  \
613152  강동구  천호역4번출구(현대백화점)  2020   6   0  1.107697e-01  1.021281e-01   
613153  강동구  천호역4번출구(현대백화점)  2020   6   1  7.194591e-02  7.115530e-02   
613154  강동구  천호역4번출구(현대백화점)  2020   6   2  4.193985e-02  4.458868e-02   
613155  강동구  천호역4번출구(현대백화점)  2020   6   3  9.366424e-02  8.890165e-02   
613156  강동구  천호역4번출구(현대백화점)  2020   6   4  1.128151e-01  1.036679e-01   
613157  강동구  천호역4번출구(현대백화점)  2020   6   5  7.362406e-04  1.145263e-03   
613158  강동구  천호역4번출구(현대백화점)  2020   6   6  2.453344e-06  5.942544e-06   
613159  강동구  천호역4번출구(현대백화점)  2020   6   7  4.259449e-07  1.176117e-06   
613160  강동구  천호역4번출구(현대백화점)  2020   6   8  1.835890e-04  3.192852e-04   
613161  강동구  천호역4번출구(현대백화점)  2020   6   9  1.652891e-03  2.404205e-03   
613162  강동구  천호역4번출구(현대백화점)  2020   6  10  8.408656e-05  1.554808e-04   
613163  강동구  천호역4번출구(현대백화점)  2020   6  11  6.376277e-02  6.416633e-02   
613164  강동구  천호역4번출구(현대백화점)  2020   6  12  1.776193e-01  1.481453e-01   
613165  강동구  천호역4번출구(현대백화점)  2020   6  13  1.161003e-01  1.061229e-01   
613166  강동구  천호역4번출구(현대백화점)  2020   6  14  5.655135e-02  5.785638e-02   
613167  강동구  천호역4번출구(현대백화점)  2020   6  15  1.892890e-01  1.552990e-01   
613168  강동구  천호역4번출구(현대백화점)  2020   6  16  1.309422e-01  1.169397e-01   
613169  강동구  천호역4번출구(현대백화점)  2020   6  17  1.157107e-01  1.058329e-01   
613170  강동구  천호역4번출구(현대백화점)  2020   6  18  1.719697e-01  1.445899e-01   
613171  강동구  천호역4번출구(현대백화점)  2020   6  19  9.089658e-02  8.670136e-02   
613172  강동구  천호역4번출구(현대백화점)  2020   6  20  7.052193e-02  6.995167e-02   
613173  강동구  천호역4번출구(현대백화점)  2020   6  21  6.204890e-02  6.267991e-02   
613174  강동구  천호역4번출구(현대백화점)  2020   6  22  8.336404e-02  8.062437e-02   
613175  강동구  천호역4번출구(현대백화점)  2020   6  23  2.901041e-02  3.220840e-02   
613176  강동구  천호역4번출구(현대백화점)  2020   7   0  6.666667e-02  6.666667e-02   
613177  강동구  천호역4번출구(현대백화점)  2020   7   1  7.969645e-02  7.761741e-02   
613178  강동구  천호역4번출구(현대백화점)  2020   7   2  5.782547e-02  5.898198e-02   
613179  강동구  천호역4번출구(현대백화점)  2020   7   3  7.693381e-02  7.533102e-02   
613180  강동구  천호역4번출구(현대백화점)  2020   7   4  2.029441e-01  1.633453e-01   
613181  강동구  천호역4번출구(현대백화점)  2020   7   5  2.880300e-03  3.995254e-03   
613182  강동구  천호역4번출구(현대백화점)  2020   7   6  1.898006e-06  4.686913e-06   
613183  강동구  천호역4번출구(현대백화점)  2020   7   7  2.771038e-03  3.856599e-03   
613184  강동구  천호역4번출구(현대백화점)  2020   7   8  7.028051e-03  9.001682e-03   
613185  강동구  천호역4번출구(현대백화점)  2020   7   9  4.911057e-02  5.118566e-02   
613186  강동구  천호역4번출구(현대백화점)  2020   7  10  5.410439e-03  7.097565e-03   
613187  강동구  천호역4번출구(현대백화점)  2020   7  11  3.111399e-02  3.427048e-02   
613188  강동구  천호역4번출구(현대백화점)  2020   7  12  1.259273e-01  1.133346e-01   
613189  강동구  천호역4번출구(현대백화점)  2020   7  13  2.929373e-01  2.076158e-01   
613190  강동구  천호역4번출구(현대백화점)  2020   7  14  6.393284e-02  6.431340e-02   
613191  강동구  천호역4번출구(현대백화점)  2020   7  15  6.298382e-02  6.349175e-02   
613192  강동구  천호역4번출구(현대백화점)  2020   7  16  1.032453e-01  9.638771e-02   
613193  강동구  천호역4번출구(현대백화점)  2020   7  17  1.618901e-01  1.380965e-01   
613194  강동구  천호역4번출구(현대백화점)  2020   7  18  8.343613e-02  8.068316e-02   
613195  강동구  천호역4번출구(현대백화점)  2020   7  19  5.702867e-02  5.827862e-02   
613196  강동구  천호역4번출구(현대백화점)  2020   7  20  1.147101e-01  1.050868e-01   
613197  강동구  천호역4번출구(현대백화점)  2020   7  21  8.221794e-02  7.968816e-02   
613198  강동구  천호역4번출구(현대백화점)  2020   7  22  3.701193e-02  3.994939e-02   
613199  강동구  천호역4번출구(현대백화점)  2020   7  23  7.879654e-02  7.687468e-02   
613200  강동구  천호역4번출구(현대백화점)  2020   8   0  5.704172e-03  7.447113e-03   
613201  강동구  천호역4번출구(현대백화점)  2020   8   1  2.018295e-02  2.330989e-02   
613202  강동구  천호역4번출구(현대백화점)  2020   8   2  7.391091e-02  7.280777e-02   
613203  강동구  천호역4번출구(현대백화점)  2020   8   3  1.997254e-01  1.614801e-01   
613204  강동구  천호역4번출구(현대백화점)  2020   8   4  2.400355e-03  3.382318e-03   
613205  강동구  천호역4번출구(현대백화점)  2020   8   5  3.640501e-01  2.316683e

In [5]:
data = pd.read_csv('./천호역4번출구(현대백화점) 균형, 불균형.csv', encoding = 'cp949')
data

자치구            대여소명     년   월  시간        재고량     3사분위수     1사분위수   균형
0    강동구  천호역4번출구(현대백화점)  2020   6   0   4.800000  7.158333  2.566667   균형
1    강동구  천호역4번출구(현대백화점)  2020   6   1   4.833333  7.158333  2.566667   균형
2    강동구  천호역4번출구(현대백화점)  2020   6   2   5.400000  7.158333  2.566667   균형
3    강동구  천호역4번출구(현대백화점)  2020   6   3   5.400000  7.158333  2.566667   균형
4    강동구  천호역4번출구(현대백화점)  2020   6   4   5.533333  7.158333  2.566667   균형
5    강동구  천호역4번출구(현대백화점)  2020   6   5   6.266667  7.158333  2.566667   균형
6    강동구  천호역4번출구(현대백화점)  2020   6   6   8.466667  7.158333  2.566667  불균형
7    강동구  천호역4번출구(현대백화점)  2020   6   7  12.466667  7.158333  2.566667  불균형
8    강동구  천호역4번출구(현대백화점)  2020   6   8   9.400000  7.158333  2.566667  불균형
9    강동구  천호역4번출구(현대백화점)  2020   6   9  11.733333  7.158333  2.566667  불균형
10   강동구  천호역4번출구(현대백화점)  2020   6  10  14.733333  7.158333  2.566667  불균형
11   강동구  천호역4번출구(현대백화점)  2020   6  11  14.766667  7.158333  2.566667  불균형
12   강동구  천호역4번출구(현대백화점)  2020   6  12  13.566667  7.158333  2.566667  불균형
13   강동구  천호역4번출구(현대백화점)  2020   6  13  12.833333  7.158333  2.566667  불균형
14   강동구  천호역4번출구(현대백화점)  2020   6  14  13.033333  7.158333  2.566667  불균형
15   강동구  천호역4번출구(현대백화점)  2020   6  15  11.100000  7.158333  2.566667  불균형
16   강동구  천호역4번출구(현대백화점)  2020   6  16   9.866667  7.158333  2.566667  불균형
17   강동구  천호역4번출구(현대백화점)  2020   6  17   8.700000  7.158333  2.566667  불균형
18   강동구  천호역4번출구(현대백화점)  2020   6  18   5.766667  7.158333  2.566667   균형
19   강동구  천호역4번출구(현대백화점)  2020   6  19   5.066667  7.158333  2.566667   균형
20   강동구  천호역4번출구(현대백화점)  2020   6  20   5.000000  7.158333  2.566667   균형
21   강동구  천호역4번출구(현대백화점)  2020   6  21   5.100000  7.158333  2.566667   균형
22   강동구  천호역4번출구(현대백화점)  2020   6  22   4.866667  7.158333  2.566667   균형
23   강동구  천호역4번출구(현대백화점)  2020   6  23   5.400000  7.158333  2.566667   균형
24   강동구  천호역4번출구(현대백화점)  2020   7   0   2.000000  7.158333  2.566667  불균형
25   강동구  천호역4번출구(현대백화점)  2020   7   1   1.900000  7.158333  2.566667  불균형
26   강동구  천호역4번출구(현대백화점)  2020   7   2   2.066667  7.158333  2.566667  불균형
27   강동구  천호역4번출구(현대백화점)  2020   7   3   2.233333  7.158333  2.566667  불균형
28   강동구  천호역4번출구(현대백화점)  2020   7   4   2.366667  7.158333  2.566667  불균형
29   강동구  천호역4번출구(현대백화점)  2020   7   5   3.033333  7.158333  2.566667   균형
30   강동구  천호역4번출구(현대백화점)  2020   7   6   5.433333  7.158333  2.566667   균형
31   강동구  천호역4번출구(현대백화점)  2020   7   7   6.700000  7.158333  2.566667   균형
32   강동구  천호역4번출구(현대백화점)  2020   7   8   5.200000  7.158333  2.566667   균형
33   강동구  천호역4번출구(현대백화점)  2020   7   9   5.433333  7.158333  2.566667   균형
34   강동구  천호역4번출구(현대백화점)  2020   7  10   6.400000  7.158333  2.566667   균형
35   강동구  천호역4번출구(현대백화점)  2020   7  11   6.900000  7.158333  2.566667   균형
36   강동구  천호역4번출구(현대백화점)  2020   7  12   6.466667  7.158333  2.566667   균형
37   강동구  천호역4번출구(현대백화점)  2020   7  13   4.633333  7.158333  2.566667   균형
38   강동구  천호역4번출구(현대백화점)  2020   7  14   4.700000  7.158333  2.566667   균형
39   강동구  천호역4번출구(현대백화점)  2020   7  15   4.833333  7.158333  2.566667   균형
40   강동구  천호역4번출구(현대백화점)  2020   7  16   4.233333  7.158333  2.566667   균형
41   강동구  천호역4번출구(현대백화점)  2020   7  17   2.533333  7.158333  2.566667  불균형
42   강동구  천호역4번출구(현대백화점)  2020   7  18   2.133333  7.158333  2.566667  불균형
43   강동구  천호역4번출구(현대백화점)  2020   7  19   2.400000  7.158333  2.566667  불균형
44   강동구  천호역4번출구(현대백화점)  2020   7  20   1.566667  7.158333  2.566667  불균형
45   강동구  천호역4번출구(현대백화점)  2020   7  21   1.400000  7.158333  2.566667  불균형
46   강동구  천호역4번출구(현대백화점)  2020   7  22   1.933333  7.158333  2.566667  불균형
47   강동구  천호역4번출구(현대백화점)  2020   7  23   1.800000  7.158333  2.566667  불균형
48   강동구  천호역4번출구(현대백화점)  2020   8   0   3.300000  7.158333  2.566667   균형
49   강동구  천호역4번출구(현대백화점)  2020   8   1   3.700000  7.158333  2.566667   균형
50   강동구  천호역4번출구(현대백화점)  2020   8   2   3.766667  7.158333  2.566667   균형
51   강동구  천호역4번출구(현대백화점)  2020   8   3   3.733333  7.158333  2.566667   균형
52   강동구  천호역4번출구(현대백화점)  2020

In [6]:
data = data.drop(['3사분위수', '1사분위수'], axis = 1)
data

자치구            대여소명     년   월  시간        재고량   균형
0    강동구  천호역4번출구(현대백화점)  2020   6   0   4.800000   균형
1    강동구  천호역4번출구(현대백화점)  2020   6   1   4.833333   균형
2    강동구  천호역4번출구(현대백화점)  2020   6   2   5.400000   균형
3    강동구  천호역4번출구(현대백화점)  2020   6   3   5.400000   균형
4    강동구  천호역4번출구(현대백화점)  2020   6   4   5.533333   균형
5    강동구  천호역4번출구(현대백화점)  2020   6   5   6.266667   균형
6    강동구  천호역4번출구(현대백화점)  2020   6   6   8.466667  불균형
7    강동구  천호역4번출구(현대백화점)  2020   6   7  12.466667  불균형
8    강동구  천호역4번출구(현대백화점)  2020   6   8   9.400000  불균형
9    강동구  천호역4번출구(현대백화점)  2020   6   9  11.733333  불균형
10   강동구  천호역4번출구(현대백화점)  2020   6  10  14.733333  불균형
11   강동구  천호역4번출구(현대백화점)  2020   6  11  14.766667  불균형
12   강동구  천호역4번출구(현대백화점)  2020   6  12  13.566667  불균형
13   강동구  천호역4번출구(현대백화점)  2020   6  13  12.833333  불균형
14   강동구  천호역4번출구(현대백화점)  2020   6  14  13.033333  불균형
15   강동구  천호역4번출구(현대백화점)  2020   6  15  11.100000  불균형
16   강동구  천호역4번출구(현대백화점)  2020   6  16   9.866667  불균형
17   강동구  천호역4번출구(현대백화점)  2020   6  17   8.700000  불균형
18   강동구  천호역4번출구(현대백화점)  2020   6  18   5.766667   균형
19   강동구  천호역4번출구(현대백화점)  2020   6  19   5.066667   균형
20   강동구  천호역4번출구(현대백화점)  2020   6  20   5.000000   균형
21   강동구  천호역4번출구(현대백화점)  2020   6  21   5.100000   균형
22   강동구  천호역4번출구(현대백화점)  2020   6  22   4.866667   균형
23   강동구  천호역4번출구(현대백화점)  2020   6  23   5.400000   균형
24   강동구  천호역4번출구(현대백화점)  2020   7   0   2.000000  불균형
25   강동구  천호역4번출구(현대백화점)  2020   7   1   1.900000  불균형
26   강동구  천호역4번출구(현대백화점)  2020   7   2   2.066667  불균형
27   강동구  천호역4번출구(현대백화점)  2020   7   3   2.233333  불균형
28   강동구  천호역4번출구(현대백화점)  2020   7   4   2.366667  불균형
29   강동구  천호역4번출구(현대백화점)  2020   7   5   3.033333   균형
30   강동구  천호역4번출구(현대백화점)  2020   7   6   5.433333   균형
31   강동구  천호역4번출구(현대백화점)  2020   7   7   6.700000   균형
32   강동구  천호역4번출구(현대백화점)  2020   7   8   5.200000   균형
33   강동구  천호역4번출구(현대백화점)  2020   7   9   5.433333   균형
34   강동구  천호역4번출구(현대백화점)  2020   7  10   6.400000   균형
35   강동구  천호역4번출구(현대백화점)  2020   7  11   6.900000   균형
36   강동구  천호역4번출구(현대백화점)  2020   7  12   6.466667   균형
37   강동구  천호역4번출구(현대백화점)  2020   7  13   4.633333   균형
38   강동구  천호역4번출구(현대백화점)  2020   7  14   4.700000   균형
39   강동구  천호역4번출구(현대백화점)  2020   7  15   4.833333   균형
40   강동구  천호역4번출구(현대백화점)  2020   7  16   4.233333   균형
41   강동구  천호역4번출구(현대백화점)  2020   7  17   2.533333  불균형
42   강동구  천호역4번출구(현대백화점)  2020   7  18   2.133333  불균형
43   강동구  천호역4번출구(현대백화점)  2020   7  19   2.400000  불균형
44   강동구  천호역4번출구(현대백화점)  2020   7  20   1.566667  불균형
45   강동구  천호역4번출구(현대백화점)  2020   7  21   1.400000  불균형
46   강동구  천호역4번출구(현대백화점)  2020   7  22   1.933333  불균형
47   강동구  천호역4번출구(현대백화점)  2020   7  23   1.800000  불균형
48   강동구  천호역4번출구(현대백화점)  2020   8   0   3.300000   균형
49   강동구  천호역4번출구(현대백화점)  2020   8   1   3.700000   균형
50   강동구  천호역4번출구(현대백화점)  2020   8   2   3.766667   균형
51   강동구  천호역4번출구(현대백화점)  2020   8   3   3.733333   균형
52   강동구  천호역4번출구(현대백화점)  2020   8   4   4.333333   균형
53   강동구  천호역4번출구(현대백화점)  2020   8   5   4.300000   균형
54   강동구  천호역4번출구(현대백화점)  2020   8   6   5.900000   균형
55   강동구  천호역4번출구(현대백화점)  2020   8   7   6.066667   균형
56   강동구  천호역4번출구(현대백화점)  2020   8   8   3.666667   균형
57   강동구  천호역4번출구(현대백화점)  2020   8   9   4.133333   균형
58   강동구  천호역4번출구(현대백화점)  2020   8  10   4.166667   균형
59   강동구  천호역4번출구(현대백화점)  2020   8  11   4.933333   균형
60   강동구  천호역4번출구(현대백화점)  2020   8  12   4.666667   균형
61   강동구  천호역4번출구(현대백화점)  2020   8  13   4.100000   균형
62   강동구  천호역4번출구(현대백화점)  2020   8  14   4.600000   균형
63   강동구  천호역4번출구(현대백화점)  2020   8  15   3.933333   균형
64   강동구  천호역4번출구(현대백화점)  2020   8  16   3.400000   균형
65   강동구  천호역4번출구(현대백화점)  2020   8  17   3.133333   균형
66   강동구  천호역4번출구(현대백화점)  2020   8  18   2.866667   균형
67   강동구  천호역4번출구(현대백화점)  2020   8  19   2.733333   균형
68   강동구  천호역4번출구(현대백화점)  2020   8  20   2.033333  불균형
69   강동구  천호역4번출구(현대백화점)  2020   8  21   2.166667  불균형
70   강동구  천호역4번출구(현대백화점)  2020   8  22   2.500000  불균형
71   강동구  천호역4번출구(현대백화점)  2020   8  23   2.56

In [7]:
df_new = pd.merge(df,data, on = ['자치구', '대여소명','년', '월', '시간'],how='outer')
df_new

자치구            대여소명     년   월  시간           p_0           p_1        L_s  \
0    강동구  천호역4번출구(현대백화점)  2020   6   0  1.107697e-01  1.021281e-01   5.520187   
1    강동구  천호역4번출구(현대백화점)  2020   6   1  7.194591e-02  7.115530e-02   6.793831   
2    강동구  천호역4번출구(현대백화점)  2020   6   2  4.193985e-02  4.458868e-02   8.127381   
3    강동구  천호역4번출구(현대백화점)  2020   6   3  9.366424e-02  8.890165e-02   6.035715   
4    강동구  천호역4번출구(현대백화점)  2020   6   4  1.128151e-01  1.036679e-01   5.462539   
5    강동구  천호역4번출구(현대백화점)  2020   6   5  7.362406e-04  1.145263e-03  12.219878   
6    강동구  천호역4번출구(현대백화점)  2020   6   6  2.453344e-06  5.942544e-06  13.296901   
7    강동구  천호역4번출구(현대백화점)  2020   6   7  4.259449e-07  1.176117e-06  13.432207   
8    강동구  천호역4번출구(현대백화점)  2020   6   8  1.835890e-04  3.192852e-04  12.650785   
9    강동구  천호역4번출구(현대백화점)  2020   6   9  1.652891e-03  2.404205e-03  11.854545   
10   강동구  천호역4번출구(현대백화점)  2020   6  10  8.408656e-05  1.554808e-04  12.823708   
11   강동구  천호역4번출구(현대백화점)  2020   6  11  6.376277e-02  6.416633e-02   7.117754   
12   강동구  천호역4번출구(현대백화점)  2020   6  12  1.776193e-01  1.481453e-01   3.970467   
13   강동구  천호역4번출구(현대백화점)  2020   6  13  1.161003e-01  1.061229e-01   5.371583   
14   강동구  천호역4번출구(현대백화점)  2020   6  14  5.655135e-02  5.785638e-02   7.425041   
15   강동구  천호역4번출구(현대백화점)  2020   6  15  1.892890e-01  1.552990e-01   3.756806   
16   강동구  천호역4번출구(현대백화점)  2020   6  16  1.309422e-01  1.169397e-01   4.984059   
17   강동구  천호역4번출구(현대백화점)  2020   6  17  1.157107e-01  1.058329e-01   5.382267   
18   강동구  천호역4번출구(현대백화점)  2020   6  18  1.719697e-01  1.445899e-01   4.079034   
19   강동구  천호역4번출구(현대백화점)  2020   6  19  9.089658e-02  8.670136e-02   6.125277   
20   강동구  천호역4번출구(현대백화점)  2020   6  20  7.052193e-02  6.995167e-02   6.848481   
21   강동구  천호역4번출구(현대백화점)  2020   6  21  6.204890e-02  6.267991e-02   7.188799   
22   강동구  천호역4번출구(현대백화점)  2020   6  22  8.336404e-02  8.062437e-02   6.378843   
23   강동구  천호역4번출구(현대백화점)  2020   6  23  2.901041e-02  3.220840e-02   8.876059   
24   강동구  천호역4번출구(현대백화점)  2020   7   0  6.666667e-02  6.666667e-02   7.000000   
25   강동구  천호역4번출구(현대백화점)  2020   7   1  7.969645e-02  7.761741e-02   6.507873   
26   강동구  천호역4번출구(현대백화점)  2020   7   2  5.782547e-02  5.898198e-02   7.369104   
27   강동구  천호역4번출구(현대백화점)  2020   7   3  7.693381e-02  7.533102e-02   6.607658   
28   강동구  천호역4번출구(현대백화점)  2020   7   4  2.029441e-01  1.633453e-01   3.523670   
29   강동구  천호역4번출구(현대백화점)  2020   7   5  2.880300e-03  3.995254e-03  11.528278   
30   강동구  천호역4번출구(현대백화점)  2020   7   6  1.898006e-06  4.686913e-06  13.319464   
31   강동구  천호역4번출구(현대백화점)  2020   7   7  2.771038e-03  3.856599e-03  11.553470   
32   강동구  천호역4번출구(현대백화점)  2020   7   8  7.028051e-03  9.001682e-03  10.814425   
33   강동구  천호역4번출구(현대백화점)  2020   7   9  4.911057e-02  5.118566e-02   7.767586   
34   강동구  천호역4번출구(현대백화점)  2020   7  10  5.410439e-03  7.097565e-03  11.053364   
35   강동구  천호역4번출구(현대백화점)  2020   7  11  3.111399e-02  3.427048e-02   8.743282   
36   강동구  천호역4번출구(현대백화점)  2020   7  12  1.259273e-01  1.133346e-01   5.110902   
37   강동구  천호역4번출구(현대백화점)  2020   7  13  2.929373e-01  2.076158e-01   2.347062   
38   강동구  천호역4번출구(현대백화점)  2020   7  14  6.393284e-02  6.431340e-02   7.110767   
39   강동구  천호역4번출구(현대백화점)  2020   7  15  6.298382e-02  6.349175e-02   7.149897   
40   강동구  천호역4번출구(현대백화점)  2020   7  16  1.032453e-01  9.638771e-02   5.739317   
41   강동구  천호역4번출구(현대백화점)  2020   7  17  1.618901e-01  1.380965e-01   4.281607   
42   강동구  천호역4번출구(현대백화점)  2020   7  18  8.343613e-02  8.068316e-02   6.376344   
43   강동구  천호역4번출구(현대백화점)  2020   7  19  5.702867e-02  5.827862e-02   7.403998   
44   강동구  천호역4번출구(현대백화점)  2020   7  20  1.147101e-01  1.050868e-01   5.409830   
45   강동구  천호역4번출구(현대백화점)  2020   7  21  8.221794e-02  7.968816e-02   6.418754   
46   강동구  천호역4번출구(현대백화점)  2020   7  22  3.701193e-02  3.994939e-02   8.395255   
47   강동구  천호역4번출구(현대백화점)  2020   7  23  7.879654e-02  7.687468e-02   6.540126   
48   강동구  천호역4번출구(현대백화점)  2020   8  

In [8]:
len(df_new)

288

In [9]:
# 재고량이 ls보다 많거나 같을 경우 불균형 아님 = 균형임

z = df_new[df_new['L_s'] <= df_new['재고량']]
z

자치구            대여소명     년   월  시간       p_0       p_1        L_s  \
4    강동구  천호역4번출구(현대백화점)  2020   6   4  0.112815  0.103668   5.462539   
10   강동구  천호역4번출구(현대백화점)  2020   6  10  0.000084  0.000155  12.823708   
11   강동구  천호역4번출구(현대백화점)  2020   6  11  0.063763  0.064166   7.117754   
12   강동구  천호역4번출구(현대백화점)  2020   6  12  0.177619  0.148145   3.970467   
13   강동구  천호역4번출구(현대백화점)  2020   6  13  0.116100  0.106123   5.371583   
14   강동구  천호역4번출구(현대백화점)  2020   6  14  0.056551  0.057856   7.425041   
15   강동구  천호역4번출구(현대백화점)  2020   6  15  0.189289  0.155299   3.756806   
16   강동구  천호역4번출구(현대백화점)  2020   6  16  0.130942  0.116940   4.984059   
17   강동구  천호역4번출구(현대백화점)  2020   6  17  0.115711  0.105833   5.382267   
18   강동구  천호역4번출구(현대백화점)  2020   6  18  0.171970  0.144590   4.079034   
36   강동구  천호역4번출구(현대백화점)  2020   7  12  0.125927  0.113335   5.110902   
37   강동구  천호역4번출구(현대백화점)  2020   7  13  0.292937  0.207616   2.347062   
51   강동구  천호역4번출구(현대백화점)  2020   8   3  0.199725  0.161480   3.577064   
53   강동구  천호역4번출구(현대백화점)  2020   8   5  0.364050  0.231668   1.732933   
61   강동구  천호역4번출구(현대백화점)  2020   8  13  0.196211  0.159422   3.636443   
75   강동구  천호역4번출구(현대백화점)  2020   9   3  0.334096  0.222731   1.965667   
77   강동구  천호역4번출구(현대백화점)  2020   9   5  0.334096  0.222731   1.965667   
84   강동구  천호역4번출구(현대백화점)  2020   9  12  0.132231  0.117858   4.952094   
85   강동구  천호역4번출구(현대백화점)  2020   9  13  0.194322  0.158305   3.668842   
99   강동구  천호역4번출구(현대백화점)  2020  10   3  0.431126  0.245296   1.316820   
101  강동구  천호역4번출구(현대백화점)  2020  10   5  0.334096  0.222731   1.965667   
107  강동구  천호역4번출구(현대백화점)  2020  10  11  0.121624  0.110201   5.223024   
108  강동구  천호역4번출구(현대백화점)  2020  10  12  0.153374  0.132459   4.462184   
109  강동구  천호역4번출구(현대백화점)  2020  10  13  0.210722  0.167772   3.398434   
124  강동구  천호역4번출구(현대백화점)  2020  11   4  0.085681  0.082507   6.299265   
125  강동구  천호역4번출구(현대백화점)  2020  11   5  0.535720  0.248727   0.866516   
129  강동구  천호역4번출구(현대백화점)  2020  11   9  0.003031  0.004186  11.494358   
130  강동구  천호역4번출구(현대백화점)  2020  11  10  0.012541  0.015201  10.172557   
131  강동구  천호역4번출구(현대백화점)  2020  11  11  0.100092  0.093946   5.834616   
132  강동구  천호역4번출구(현대백화점)  2020  11  12  0.246879  0.186864   2.880058   
133  강동구  천호역4번출구(현대백화점)  2020  11  13  0.168835  0.142591   4.140796   
134  강동구  천호역4번출구(현대백화점)  2020  11  14  0.031195  0.034349   8.738298   
135  강동구  천호역4번출구(현대백화점)  2020  11  15  0.068834  0.068518   6.914183   
136  강동구  천호역4번출구(현대백화점)  2020  11  16  0.134012  0.119122   4.908350   
137  강동구  천호역4번출구(현대백화점)  2020  11  17  0.103402  0.096508   5.734650   
138  강동구  천호역4번출구(현대백화점)  2020  11  18  0.104869  0.097636   5.691073   
139  강동구  천호역4번출구(현대백화점)  2020  11  19  0.222726  0.174383   3.215158   
140  강동구  천호역4번출구(현대백화점)  2020  11  20  0.176634  0.147530   3.989146   
141  강동구  천호역4번출구(현대백화점)  2020  11  21  0.184413  0.152341   3.844400   
147  강동구  천호역4번출구(현대백화점)  2020  12   3  0.750000  0.187500   0.333333   
170  강동구  천호역4번출구(현대백화점)  2021   1   2  0.625000  0.234375   0.599994   
173  강동구  천호역4번출구(현대백화점)  2021   1   5  0.571430  0.244899   0.749955   
183  강동구  천호역4번출구(현대백화점)  2021   1  15  0.169505  0.143020   4.127490   
186  강동구  천호역4번출구(현대백화점)  2021   1  18  0.197509  0.160185   3.614378   
187  강동구  천호역4번출구(현대백화점)  2021   1  19  0.384197  0.236693   1.594159   
188  강동구  천호역4번출구(현대백화점)  2021   1  20  0.578949  0.243768   0.727238   
189  강동구  천호역4번출구(현대백화점)  2021   1  21  0.290636  0.206675   2.370804   
194  강동구  천호역4번출구(현대백화점)  2021   2   2  0.334096  0.222731   1.965667   
204  강동구  천호역4번출구(현대백화점)  2021   2  12  0.040008  0.042780   8.230206   
205  강동구  천호역4번출구(현대백화점)  2021   2  13  0.264338  0.195203   2.662971   
206  강동구  천호역4번출구(현대백화점)  2021   2  14  0.216112  0.170774   3.314686   
207  강동구  천호역4번출구(현대백화점)  2021   2  15  0.127019  0.114124   5.082948   
210  강동구  천호역4번출구(현대백화점)  2021   2  18  0.144038  0.126119   4.670839   
211  강동구  천호역4번출구(현대백화점)  2021   2  19  0.190750  0.156177   3.

In [10]:
len(z)

95

In [11]:
# TP

len(z[z['균형'] == '균형'])

39

In [12]:
# FN

len(z[z['균형'] == '불균형'])

56

- 재고량이 ls보다 많거나 같을 경우 균형인데 불균형이라고 판단한 수가 56개

- 균형인데 균형이라고 판단한 수가 39개

In [13]:
# ls보다 재고량이 작으므로 불균형

w = df_new[df_new['L_s'] > df_new['재고량']]
w

자치구            대여소명     년   월  시간           p_0           p_1        L_s  \
0    강동구  천호역4번출구(현대백화점)  2020   6   0  1.107697e-01  1.021281e-01   5.520187   
1    강동구  천호역4번출구(현대백화점)  2020   6   1  7.194591e-02  7.115530e-02   6.793831   
2    강동구  천호역4번출구(현대백화점)  2020   6   2  4.193985e-02  4.458868e-02   8.127381   
3    강동구  천호역4번출구(현대백화점)  2020   6   3  9.366424e-02  8.890165e-02   6.035715   
5    강동구  천호역4번출구(현대백화점)  2020   6   5  7.362406e-04  1.145263e-03  12.219878   
6    강동구  천호역4번출구(현대백화점)  2020   6   6  2.453344e-06  5.942544e-06  13.296901   
7    강동구  천호역4번출구(현대백화점)  2020   6   7  4.259449e-07  1.176117e-06  13.432207   
8    강동구  천호역4번출구(현대백화점)  2020   6   8  1.835890e-04  3.192852e-04  12.650785   
9    강동구  천호역4번출구(현대백화점)  2020   6   9  1.652891e-03  2.404205e-03  11.854545   
19   강동구  천호역4번출구(현대백화점)  2020   6  19  9.089658e-02  8.670136e-02   6.125277   
20   강동구  천호역4번출구(현대백화점)  2020   6  20  7.052193e-02  6.995167e-02   6.848481   
21   강동구  천호역4번출구(현대백화점)  2020   6  21  6.204890e-02  6.267991e-02   7.188799   
22   강동구  천호역4번출구(현대백화점)  2020   6  22  8.336404e-02  8.062437e-02   6.378843   
23   강동구  천호역4번출구(현대백화점)  2020   6  23  2.901041e-02  3.220840e-02   8.876059   
24   강동구  천호역4번출구(현대백화점)  2020   7   0  6.666667e-02  6.666667e-02   7.000000   
25   강동구  천호역4번출구(현대백화점)  2020   7   1  7.969645e-02  7.761741e-02   6.507873   
26   강동구  천호역4번출구(현대백화점)  2020   7   2  5.782547e-02  5.898198e-02   7.369104   
27   강동구  천호역4번출구(현대백화점)  2020   7   3  7.693381e-02  7.533102e-02   6.607658   
28   강동구  천호역4번출구(현대백화점)  2020   7   4  2.029441e-01  1.633453e-01   3.523670   
29   강동구  천호역4번출구(현대백화점)  2020   7   5  2.880300e-03  3.995254e-03  11.528278   
30   강동구  천호역4번출구(현대백화점)  2020   7   6  1.898006e-06  4.686913e-06  13.319464   
31   강동구  천호역4번출구(현대백화점)  2020   7   7  2.771038e-03  3.856599e-03  11.553470   
32   강동구  천호역4번출구(현대백화점)  2020   7   8  7.028051e-03  9.001682e-03  10.814425   
33   강동구  천호역4번출구(현대백화점)  2020   7   9  4.911057e-02  5.118566e-02   7.767586   
34   강동구  천호역4번출구(현대백화점)  2020   7  10  5.410439e-03  7.097565e-03  11.053364   
35   강동구  천호역4번출구(현대백화점)  2020   7  11  3.111399e-02  3.427048e-02   8.743282   
38   강동구  천호역4번출구(현대백화점)  2020   7  14  6.393284e-02  6.431340e-02   7.110767   
39   강동구  천호역4번출구(현대백화점)  2020   7  15  6.298382e-02  6.349175e-02   7.149897   
40   강동구  천호역4번출구(현대백화점)  2020   7  16  1.032453e-01  9.638771e-02   5.739317   
41   강동구  천호역4번출구(현대백화점)  2020   7  17  1.618901e-01  1.380965e-01   4.281607   
42   강동구  천호역4번출구(현대백화점)  2020   7  18  8.343613e-02  8.068316e-02   6.376344   
43   강동구  천호역4번출구(현대백화점)  2020   7  19  5.702867e-02  5.827862e-02   7.403998   
44   강동구  천호역4번출구(현대백화점)  2020   7  20  1.147101e-01  1.050868e-01   5.409830   
45   강동구  천호역4번출구(현대백화점)  2020   7  21  8.221794e-02  7.968816e-02   6.418754   
46   강동구  천호역4번출구(현대백화점)  2020   7  22  3.701193e-02  3.994939e-02   8.395255   
47   강동구  천호역4번출구(현대백화점)  2020   7  23  7.879654e-02  7.687468e-02   6.540126   
48   강동구  천호역4번출구(현대백화점)  2020   8   0  5.704172e-03  7.447113e-03  11.007296   
49   강동구  천호역4번출구(현대백화점)  2020   8   1  2.018295e-02  2.330989e-02   9.499531   
50   강동구  천호역4번출구(현대백화점)  2020   8   2  7.391091e-02  7.280777e-02   6.719532   
52   강동구  천호역4번출구(현대백화점)  2020   8   4  2.400355e-03  3.382318e-03  11.643569   
54   강동구  천호역4번출구(현대백화점)  2020   8   6  1.683372e-07  4.979975e-07  13.489363   
55   강동구  천호역4번출구(현대백화점)  2020   8   7  4.246180e-02  4.507484e-02   8.100065   
56   강동구  천호역4번출구(현대백화점)  2020   8   8  1.040410e-03  1.573000e-03  12.076998   
57   강동구  천호역4번출구(현대백화점)  2020   8   9  2.327862e-02  2.647943e-02   9.266758   
58   강동구  천호역4번출구(현대백화점)  2020   8  10  6.666667e-02  6.666667e-02   7.000000   
59   강동구  천호역4번출구(현대백화점)  2020   8  11  1.289651e-02  1.558919e-02  10.137039   
60   강동구  천호역4번출구(현대백화점)  2020   8  12  1.245295e-01  1.123208e-01   5.146981   
62   강동구  천호역4번출구(현대백화점)  2020   8  14  2.625911e-02  2.947957e-02   9.057846   
63   강동구  천호역4번출구(현대백화점)  2020   8  

In [14]:
len(w)

193

In [15]:
# FP

len(w[w['균형'] == '균형'])

106

In [16]:
# TN

len(w[w['균형'] == '불균형'])

87

- 불균형인데 균형이라고 판단한 수는 106
- 불균형인데 불균형이라고 판단한 수는 87

# confusion matrix

0 |균형|불균형
-|-|-
균형|TP|FP
불균형|FN|TN


1. 정확도 accuracy = 실제(내) 데이터와 예측(기존방법)데이터가 얼마나 같은지를 평가하는 지표 = TP + TN / TP+TN+FP+FN
2. 정밀도 precision = 균형으로 예측한 데이터 중 실제로 균형인 데이터 비율 = TP / TP+FP
3. 재현율 recall = 실제 균형 중 균형으로 예측한 비율 = TP / TP+FN

=> 정밀도에 초점. (불균형을 균형으로 예측하여 재배치하지 않으면 서비스에 차질)


In [17]:
# 정확도

(len(z[z['균형'] == '균형']) + len(w[w['균형'] == '불균형'])) / len(df_new)

0.4375

In [18]:
# 정밀도

len(z[z['균형'] == '균형']) / (len(z[z['균형'] == '균형']) + len(w[w['균형'] == '균형']))

0.2689655172413793

In [19]:
# 재현율

len(z[z['균형'] == '균형']) / (len(z[z['균형'] == '균형']) + len(z[z['균형'] == '불균형']))

0.4105263157894737